### ISSUE TITLE
Validate for duplicate alias when creating custom fields

### ISSUE THREAD

**Issue by justinfortes** [mautic/mautic#5694]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Y
| Feature request? | 
| Enhancement? | Y

## Description:
The more details the better...

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 
| PHP version | 

### Steps to reproduce:
1. Create a custom field
2. Use an alias that already exists - eg. source
3. Page allows you to create a field with a duplicate alias name
4. This field no longer can be used in a segment or campaign as the alias conflicts with the field with the same alias. 

Suggested fix:
On the create page, when hitting "save & close" or "apply", check the existing standard fields and custom fields to ensure that the alias is unique. 

When a field exists with the same alias present the red bordered box around the field with the text "Alias must be unique"

**Comment by alanhartless** [comment]
The issue isn't duplicate alias for custom fields. It's custom field aliases that conflict with non-custom field filters for segments. For example, `source` is used as the filter for page hits based query.

**Comment by escopecz** [comment]
As Alan noted, the checking for duplicate aliases is there already. So should we check for some keywords too? If so, any idea what are they?
![screen shot 2018-03-19 at 17 41 41](https://user-images.githubusercontent.com/1235442/37609315-16d13ae4-2b9d-11e8-87e5-5b45811ec9bf.png)

**Comment by escopecz** [comment]
@alanhartless those are the keywords we want to avoid? https://github.com/mautic/mautic/blob/7218b203f1b3ae974cfdf5b2889fa6389bee52f4/app/bundles/LeadBundle/Model/ListModel.php#L243

**Comment by alanhartless** [comment]
Yes. Because if we have a custom field named "sessions", the UI may treat it as a custom field filter but the backend treats it as a page hits based filter resulting in different behavior.
